---
title: Characterisation of a Putative Agent of Horizontal Gene Transfer Encoded by
  a Bacterial Prophage Region
author: "Y3877831"
output:
  bookdown::html_document2:
    css: css/bootstrap-minty.css
    toc: yes
    depth: 3
    toc_float:
      collapsed: no
      smooth_scroll: yes
    number_sections: yes
    global_numbering: yes
    highlight: null
    df_print: paged
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  include = FALSE,
  echo = FALSE,
  fig.align = "center",
  fig.retina = 3,
  message = FALSE,
  warning = FALSE)
```

```{r pkgs}
source("../R/00-pkgs.R", local = knitr::knit_global())
```

```{r themes}
font_add_google("Montserrat", "Montserrat")
# turn on showtext
showtext_auto()

source("../R/themes.R", local = knitr::knit_global())
```

```{r write-pkgs-bib}
knitr::write_bib(
  c(
    "base",
    "RefManageR",
    "tidyverse",
    "rmarkdown",
    "plotly",
    "ggpubr",
    "htmltools",
    "kableExtra",
    "showtext"
  ),
  width = 80,
  prefix = "",
  file = "../bibliography/pkgs.bib"
)
```

```{r import-bibliography}
RefManageR::BibOptions(
  check.entries = FALSE,
  bib.style = "authoryear",
  cite.style = "authoryear",
  style = "markdown",
  hyperlink = FALSE,
  dashed = FALSE,
  longnamesfirst = FALSE,
  max.names = 3
)

refs <-
  RefManageR::ReadBib("../bibliography/refs.bib", check = FALSE)
pkgs <-
  RefManageR::ReadBib("../bibliography/pkgs.bib", check = FALSE)
```
___
# Introduction
Double strand breaks (DSBs) are a form of DNA damage that results in the loss of bases on both strands of the DNA helix. They are commonly repaired by the non-homologous end-joining (NHEJ) pathway, which requires the protein Ku. Ku is a heterodimer of Ku70/Ku80 subunits. It identifies and binds to DSBs prior to NHEJ, and takes part in some transposition events `r Cite(refs, "DAdda_di_Fagagna2003-kb")`. Homologues of Ku have been found in many bacteria. The most well- known homologue of Ku is the Gam protein of the Mu bacteriophage. 

Mu is a temperate bacteriophage of Enterobacteria, particularly *Escherichia coli*. In bacteria, Mu acts as a transposable element `r Cite(refs, "Fuller2017-ex")` – meaning it can change its position within the genome and replicate by forming two replication forks `r Cite(refs, "Shapiro1979-gf")`. The Mu transposon involves 3 proteins: MuA, MuB, and MuGam `r Cite(refs, c("DAdda_di_Fagagna2003-kb", "Fuller2017-ex"))`. MuB polymerises adjacent to the *E. coli* target site and associates with MuA. MuA binds specific target sites and synapses them together to form the transposon. MuA remains bound until it is disassembled by the ATP-ClpX chaperone `r Cite(refs, "Harshey2012-qe")`. Meanwhile, MuGam binds to excised Mu phage DNA and protects it from digestion by host nucleases. Transposition events promote the uptake of foreign linear DNA: this function is known as natural competence and is linked to horizontal gene transfer (HGT). HGT is the transfer of foreign DNA between organisms and significantly contributes to the spread of antibiotic resistance and evolution in bacteria `r Cite(refs, "Gyles2014-wy")`.

MuGam enables natural competence in *E. coli* and is therefore a putative agent of HGT. Bacteria possessing a Gam-like prophage region should also be naturally competent. In *Haemophilus influenzae*, Gam (HiGam) inhibits digestion of Mu phage DNA by nucleases `r Cite(refs, "DAdda_di_Fagagna2003-kb")`. Through sequence alignment, we identified and cloned a putative homologue of Mu phage Gam from *Haemophilus influenzae* (HiGam). The structure – including binding affinity, oligomeric state – of HiGam was assessed in comparison to MuGam in literature. A range of assays were used to achieve this, namely electrophoretic mobility shift assay (EMSA), sodium dodecyl sulphate– polyacrylamide gel electrophoresis (SDS-PAGE) and size exclusion chromatography (SEC). Similarities to Mu will insinuate that HiGam will too function as an agent of horizontal gene transfer.

The aim of this report is to characterise HiGam by processing the data obtained from assays.

___
# Requirements
Required software and packages are outlined in the [README](README.md).

___
# The Data

```{r functions}
# Source functions
source("../functions/count_words.R", local = knitr::knit_global()) # Reproducable Wordcount
source("../functions/relative_front.R", local = knitr::knit_global()) # Calculate relative front
source("../functions/sec_pred_intercept.R", local = knitr::knit_global()) # Predicts the intercept of the SEC data
source("../functions/emsa_pred_intercept.R", local = knitr::knit_global()) # Predicts the intercept of the EMSA data
source("../functions/sec_mw.R", local = knitr::knit_global()) # SEC molecular weights
source("../functions/nls.R", local = knitr::knit_global()) # Performs the NLS using the Langmuir binding isotherm (NLS)
```

```{r data-import}
# Load the PCR data
pcr <- read.csv("../data/data-raw/pcr.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the PCR calibration data for the gDNA library
gdna_calibration <-
  read.csv("../data/data-raw/pcr_gdna_calibration.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the PCR calibration data for the cDNA library
cdna_calibration <-
  read.csv("../data/data-raw/pcr_cdna_calibration.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the EMSA data
emsa <- read.csv("../data/data-raw/emsa.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the EMSA calibration data
emsa_calibration <-
  read.csv("../data/data-raw/emsa_calibration.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the SEC standards data
standards <-
  read.csv("../data/data-raw/sec_std.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the SEC data
sec <-
  read.csv("../data/data-raw/sec_im9higam.csv", header = TRUE) %>%
  janitor::clean_names()

```

```{r data-tidying}
# Rename the proteins and pivot them
standards <- standards %>%
  rename(
    c(
      "Ferritin" = curve_ferritin,
      "IgG" = curve_ig_g,
      "Myoglobin" = curve_myoglobin,
      "Ovalbumin" = curve_ovalbumin,
      "Thyroglobulin" = curve_thyroglobulin,
      "Transferrin" = curve_transferrin,
      "Vitamin B12" = curve_vitamin_b12
    )
  ) %>%
  pivot_longer(cols = !c(x, vol),
               names_to = "protein",
               values_to = "absorbance_units")

# Pivot proteins
sec <- sec %>%
  pivot_longer(cols = starts_with("a"),
               names_to = "protein",
               values_to = "absorbance")

# Separate the PCR data set by DNA library for analysis
gdna <- pcr %>% # gDNA PCR data
  filter(library == "gDNA")

cdna <- pcr %>% # cDNA PCR data
  filter(library == "cDNA")
```

___
# Analysis 
## Diagnostic PCR of the Mu Prophage Region in Haemophilus influenzae Genomic DNA and cDNA {.tabset}
```{r pcr-linear-models}
#gDNA
gdna_mod <- lm(data = gdna_calibration, log ~ rf)

#cDNA
cdna_mod <- lm(data = cdna_calibration, log ~ rf)
```

```{r pcr-preperation}
# gDNA
gdna <- gdna %>%
  mutate(
    rf = relative_front(gdna),
    # calculate the Rf
    log = ((rf * gdna_mod$coefficients[2]) + gdna_mod$coefficients[1]),
    # log = (y = mx+c)
    actual_size = (10 ^ log),
    # actual gene size = 10^log
    diff = ((expected_size - actual_size) / expected_size * 100) # % difference between expected and actual sizes
  )

# cDNA
cdna <- cdna %>%
  mutate(
    rf = relative_front(cdna),
    # calculate the Rf
    log = ((rf * cdna_mod$coefficients[2]) + cdna_mod$coefficients[1]),
    # log = (y = mx+c)
    actual_size = (10 ^ log),
    # actual gene size = 10^log
    diff = ((expected_size - actual_size) / expected_size * 100) # % difference between expected and actual sizes
  )

# join gDNA & cDNA datasets
pcr_full <- full_join(gdna, cdna, by = NULL)
```
Diagnostic PCR was used to amplify the *H. influenzae homologues*, and to test for their transcription relative to the Genomic DNA (gDNA) control. Amplicon sizes were established with equation \@ref(eq:log), using DNA ladder molecular weight (MW) standards. Results identified the transcription of HiA, HiB, HiGam, HiA-HiB and HiB-HiGam in *H. influenzae* (Figure \@ref(pcr-fig)).

\begin{equation}
\Large \color{#8D9FCA} y \color{#5a5a5a} = \color{#E789C2} m \color{#66c2a4} x \color{#5a5a5a} + \color{#FFD82F} c\\
\small \text{Where } \color{#8D9FCA} y = \text{log(DNA MW in bp), } \color{#E789C2} m = \text{gradient, } \color{#66c2a4} x = \text{Rf, and } \color{#FFD82F} c = \text{y-intercept}(\#eq:log)
\end{equation}

### Expressed Amplicons {#pcr-fig}
```{r pcr-fig, include = TRUE}
# plot of the merged PCR data showing the expressed genes and the percentage difference of actual size to expected size
pcr_plot <- pcr_full %>%
  ggplot(aes(
    x = gene,
    y = diff,
    fill = library,
    text = paste(
      "Library: ",
      library,
      "<br>Amplicon: ",
      gene,
      "<br>Difference: ",
      round(diff),
      "%",
      "<br>Expected Size (bp): ",
      expected_size,
      "<br>Actual Size (bp):",
      round(actual_size)
    )
  )) + geom_col(position = "dodge") +
  geom_hline(yintercept = 0, size = 0.25) +
  mytheme() +
  scale_fill_brewer(name = "DNA Library",
                    labels = c("cDNA", "gDNA"),
                    palette = "Set3",
  ) +
  labs(y = "Percentage Difference to Expected Size",
       x = "Amplicon")

ggsave(
  "../figures/pcr_plot.pdf",
  plot = pcr_plot,
  height = 15,
  width = 20,
  unit = "cm"
)

ggplotly(pcr_plot, tooltip = c("text"), align = "center")
```

**Figure \@ref(pcr-fig):** Expression levels of HiA, HiB and HiGam ORFs in the complementary DNA (cDNA) and genomic DNA (gDNA) libraries, compared to their expected sizes. Results indicate that HiA, HiB and HiGam genes are expressed, and that poly-cistronic transcripts encoding HiA-HiB and HiB-HiGam gene pairs exist in *H. influenzae*.


### Dataset {#pcr-tab}
```{r pcr-tab, include = TRUE}
# Table of the full PCR dataset
pcr_full %>%
  select(-primer, -gel_length) %>%
  knitr::kable(
    booktabs = T,
    digits = 2,
    col.names = c(
      "Library",
      "Amplicon",
      "Distance (cm)",
      "Expected Size (bp)",
      "Relative Front (Rf)",
      "Log(Rf)",
      "Actual Size (bp)",
      "Size Difference (%)"
    )
  ) %>%
  collapse_rows(columns = 1) %>%
  kable_styling(font_size = 15, position = "center")
```

**Table \@ref(pcr-tab):**

---
## Analyse DNA-Binding Properties of HiGam in EMSA {.tabset}
```{r emsa-prep}
# prepare the PCR dataset for analysis
emsa <-
  emsa %>% mutate(rf = (relative_front(emsa))) # calculate the relative front
```

```{r emsa-diff}
# Calculate the difference between the measured rf and predicted rf value
# This is needed for the residuals
# Calculate residuals to the fit
emsa <- emsa %>%
  mutate(residuals = rf - predict(NLS(emsa)))
```

```{r kd}
kd <- summary(NLS(emsa))$coefficients[2, 1]
```

```{r emsa-footprint}
# 
emsa_calibration <-
  emsa_calibration %>% mutate(log = log(expected_size),
                              rf = relative_front(emsa_calibration))

emsa_lm <- lm(emsa_calibration$rf ~ emsa_calibration$log)

emsa_footprint <-
  (((
    max((emsa$rf) * emsa_lm$coefficients[2]) + emsa_lm$coefficients[1]
  ) - 330000) / (30626.5 / 2))

```

Electrophoretic mobility shift assay (EMSA) was used to examine whether Im9-HiGam–DNA had an affinity for closed circular DNA or linear DNA. EMSA was performed using purified, recombinant Im9-HiGam, with supercoiled plasmid plasmid DNA and linear DNA templates. A fixed concentration of DNA was used with varying concentrations of Im9-HiGam to saturate binding to DNA. Rf was calculated in R using calculation \@ref(eq:rf).

\begin{equation} 
\frac{(\text{gel length} - \text{distance})}{\text{gel length}}(\#eq:rf)
\end{equation}

Im9-HiGam was observed to bind preferentially to linear DNA with a dissociation constant ($K_D$) of approximately `r signif(kd, digits = 2)`M (Figure \@ref(emsa)). Results estimate Im9-HiGam has a binding dimeric footprint of approximately `r signif(abs(emsa_footprint), digits = 2)` base pairs.

___
### Analysis {#emsa-fig}
```{r emsa-fig, include = TRUE}
emsa_plot <- emsa %>%
  ggplot(aes(x = conc, y = rf)) +
  geom_point(shape = 20,
             size = 3) +
  stat_function(fun = emsa_pred_intercept, colour = "#66C2A4") +
  mytheme() +
  scale_y_continuous(
    expand = c(0.02, 0.02),
    limits = c(0.3, 0.7),
    breaks = seq(0.3, 0.7, by = 0.20)
  ) +
  scale_x_continuous(
    expand = c(0.02, 0.02),
    limits = c(0, 4),
    breaks = seq(0, 8, by = 1)
  ) +
  xlab(NULL) +
  ylab(expression(paste(bold(
    "Relative Front (a.U.)"
  ))))

# Plot the residuals and a line showing y = 0
emsa_res_plot <- emsa %>%
  ggplot(aes(x = conc, y = residuals)) +
  geom_point(shape = 20,
             size = 3,
             colour = "#5a5a5a") +
  geom_hline(aes(yintercept = 0),
             col = '#66C2A4',
             size = .5) +
  scale_x_continuous(
    expand = c(0.02, 0.02),
    limits = c(0, 4),
    breaks = seq(0, 8, by = 1),
    name = (expression(bold(
      atop("Protein Concentration (μM)", "Im9"["dimer"])
    )))
  ) +
  scale_y_continuous(
    expand = c(0.02, 0.02),
    limits = c(-0.1, 0.1),
    breaks = seq(-0.1, 0.1, by = 0.1),
    name = "Residual (a.U.)"
  ) + mytheme()


emsa_full <- ggarrange(
  emsa_plot,
  emsa_res_plot,
  nrow = 2,
  ncol = 1,
  widths = c(2, 1 / 2)
)

emsa_full

ggsave("../figures/emsa_plot.pdf", plot = emsa_full, 
       height = 15, 
       width = 20, unit = "cm")
```

**Figure \@ref(emsa-fig):** The binding affinity (equilibrium dissociation constant, KD) was estimated by nonlinear regression analysis of EMSA using the Langmuir Binding Isotherm.

### Dataset {#emsa-tab}
```{r emsa-tab, include = TRUE}
# EMSA Dataset
emsa %>%
  select(-gel_length) %>%
  knitr::kable(
    booktabs = T,
    digits = 2,
    col.names = c("Concentration",
                  "Distance",
                  "Relative Front",
                  "Residuals")
  ) %>%
  kable_styling(font_size = 15, position = "center")
```

**Table \@ref(emsa-tab):**

___
## Size Exclusion Chromatography (SEC) {.tabset}
```{r sec-standards}
# Prepare the standards data
sec_standards <- standards %>%
  group_by(protein) %>%
  filter(absorbance_units == max(absorbance_units)) %>% # filter by the maximum absorbency units
  summarise(vol) %>%
  ungroup() %>%
  mutate(
    mw = recode(
      # Protein MWs
      protein,
      "Thyroglobulin" = 669000,
      "Ferritin" = 440000,
      "IgG" = 150000,
      "Transferrin" = 81000,
      "Ovalbumin" = 43000,
      "Myoglobin" = 17000,
      "Vitamin B12" = 1355
    ),
    logmw = log(mw)
  )
```

```{r sec-mw}
# Getting the elution volume that corresponds to the maximum value of a curve
a280_mw <- sec_mw("a280")
a254_mw <- sec_mw("a254")

sec_footprint <- (a254_mw - a280_mw) / 660
```

Size exclusion chromatography (SEC) was used to explore the molecular weight and oligomeric state of Im9-HiGam. 7 chromatograms of known MW were plotted and a linear regression was performed to determine the relationship between elution volume and molecular weight (Figure \@ref(sec-std-fig)). The SEC data showed a peak for both wavelengths, with a MW of approximately `r format(a254_mw, scientific = FALSE)` Daltons for A254 and `r format(a280_mw, scientific = FALSE)` Daltons for A280 (Figure \@ref(sec-fig)). This gives Im9-HiGam has an estimated monomeric footprint of approximately `r format(round(sec_footprint, digits = 2), scientific = FALSE)` base pairs.

### Analysis {#sec-fig}
```{r sec-fig, include = TRUE}
sec_plot <- sec %>%
  ggplot(aes(x = vol,
             y = absorbance,
             colour = protein),
         show.legend = TRUE) +
  geom_line() +
  scale_colour_brewer(name = "Protein",
                      labels = c("a280", "a254"),
                      palette = "Set2") +
  scale_x_continuous(limits = c(13, 17)) +
  scale_y_continuous(expand = c(0, 0),
                     limits = c(0, 100)) +
  mytheme() +
  xlab("Elution Volume (mL)") +
  ylab("Absorbance (mAU)")

ggsave(
  "../figures/sec_plot.pdf",
  height = 15,
  width = 20,
  unit = "cm"
)

ggplotly(align = "center")
```
**Figure \@ref(sec-fig):**

### Calibration Plot {#sec-calib-fig}
```{r sec-calibration, include = TRUE}
sec_std_plot <- sec_standards %>%
  ggplot(mapping = aes(y = vol, x = logmw)) +
  geom_point() +
  stat_function(fun = sec_pred_intercept,
                geom = "line",
                colour = "#8D9FCA") +
  mytheme() +
  scale_x_continuous(
    expand = c(0.05, 0.05),
    limits = c(7, 14),
    breaks = seq(7, 14, by = 1),
    name = "log (MW)"
  ) + scale_y_continuous(
    expand = c(0.05, 0.05),
    limits = c(9, 22),
    breaks = seq(9, 22, by = 2),
    name = "Elution Volume (mL)"
  )
sec_std_plot

ggsave(
  "../figures/sec_std_plot.pdf",
  plot = sec_std_plot,
  height = 15,
  width = 20,
  unit = "cm"
)
```
**Figure \@ref(sec-calib-fig):**

### Standards {#sec-std-fig}
```{r standards-plot, include = TRUE}
standards_plot <- standards %>%
  ggplot(aes(x = vol,
             y = absorbance_units,
             colour = protein),
         show.legend = TRUE) +
  geom_line() +
  scale_colour_brewer(name = "Protein",
                      palette = "Set2") + # colour blind friendly
  mytheme() +
  scale_x_continuous(
    expand = c(0, 0),
    limits = c(9, 22),
    breaks = seq(9, 22, by = 1),
    name = "Elution Volume (mL)"
  ) + scale_y_continuous(
    expand = c(0, 0),
    limits = c(0, 100),
    breaks = seq(0, 100, by = 10),
    name = "A280 Absorbance Units (A.U.)"
  )

ggplotly(standards_plot)

ggsave(
  "../figures/standards_plot.pdf",
  plot = standards_plot,
  height = 15,
  width = 20,
  unit = "cm"
)
```
**Figure \@ref(sec-std-fig):**

### Dataset {#sec-data}
```{r sec-tab, include = TRUE}
sec %>%
  select(-x) %>%
  pivot_wider(names_from = "protein",
              values_from = "absorbance") %>%
  knitr::kable(
    booktabs = T,
    digits = 2,
    col.names = c("Volume",
                  "a254",
                  "a280")
  ) %>%
  collapse_rows(columns = 1) %>%
  kable_styling(font_size = 15) %>%
  scroll_box(height = "300px")
```
**Figure \@ref(sec-data):**
___

# {-}
___
**This Document:** `r count_words("report.Rmd")`\
**README:** `r count_words("../README.md")`\
**Total:** `r count_words("report.Rmd") + count_words("../README.md")`\

___
Report made with `rmarkdown` `r Cite(pkgs, c("rmarkdown","rmarkdown2018", "rmarkdown2020"))`, `RefManageR``r Cite(pkgs, "RefManageR")`.\

Theme is *Minty* by [Bootswatch](https://bootswatch.com/minty/).

___
# Bibliography {.tabset}

## Packages {-}
```{r pkgs-bib, include = TRUE, results = "asis"}
RefManageR::PrintBibliography(pkgs)
```

## References {-}
```{r refs-bib, include = TRUE, results =  "asis"}
RefManageR::PrintBibliography(refs)
```
