---
title: Characterisation of a Putative Agent of Horizontal Gene Transfer Encoded by
  a Bacterial Prophage Region
author: "Y3877831"
output:
  bookdown::html_document2:
    css: css/bootstrap-minty.css
    toc: yes
    depth: 3
    toc_float:
      collapsed: no
      smooth_scroll: yes
    number_sections: yes
    highlight: null
    df_print: paged
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    include = FALSE,
    echo = FALSE,
    fig.align = "center",
    fig.retina = 3,
    message = FALSE,
    warning = FALSE
)
```

```{r pkgs}
source("../R/00-pkgs.R", local = knitr::knit_global())
```

```{r write-pkgs-bib}
knitr::write_bib(
    c(
        "base",
        "RefManageR",
        "tidyverse",
        "rmarkdown",
        "plotly",
        "ggpubr",
        "htmltools",
        "kableExtra"
    ),
    width = 80,
    prefix = "",
    file = "../bibliography/pkgs.bib"
)
```

```{r import-bibliography}
RefManageR::BibOptions(
    check.entries = FALSE,
    bib.style = "authoryear",
    cite.style = "authoryear",
    style = "markdown",
    hyperlink = FALSE,
    dashed = FALSE,
    longnamesfirst = FALSE,
    max.names = 3
)

refs <-
    RefManageR::ReadBib("../bibliography/refs.bib", check = FALSE)
pkgs <-
    RefManageR::ReadBib("../bibliography/pkgs.bib", check = FALSE)
```
___
# Introduction

MuGam is a orthologue of eukaryotic proteins Ku70/Ku80, which are involved in recognizing double-strand DNA prior to repair, as well as DNA transposition. In *Haemophilus influenzae*, Gam (HiGam) inhibits digestion of Mu phage DNA by nucleases `r Cite(refs, "DAdda_di_Fagagna2003-kb")`.

The aim of this report is to characterise HiGam by processing the data obtained from the following experiments:

* Polymerase chain reaction (PCR)
* Electrophoretic mobility shift assay (EMSA)
* Size-exclusion chromatography (SEC)
* Analytical ultracentrifugation (AUC) ???


___
# Requirements
Required software and packages are outlined in the [README](README.md).

# Preparing Data
```{r ggplot-themes}
source("../R/themes.R", local = knitr::knit_global())
```

```{r functions}
source("../functions/count_words.R", local = knitr::knit_global())
source("../functions/relative_front.R", local = knitr::knit_global())
source("../functions/pred_intercept_sec.R", local = knitr::knit_global())
source("../functions/pred_intercept_emsa.R", local = knitr::knit_global())
source("../functions/nls.R", local = knitr::knit_global())
```

```{r data-import}
# Load the PCR data
pcr <- read.csv("../data/data-raw/pcr.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the PCR calibration data for the gDNA library
gdna_calibration <-
  read.csv("../data/data-raw/pcr_gdna_calibration.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the PCR calibration data for the cDNA library
cdna_calibration <-
  read.csv("../data/data-raw/pcr_cdna_calibration.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the EMSA data
emsa <- read.csv("../data/data-raw/emsa.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the EMSA calibration data
emsa_calibration <-
  read.csv("../data/data-raw/emsa_calibration.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the SEC standards data
standards <-
  read.csv("../data/data-raw/sec_std.csv", header = TRUE) %>%
  janitor::clean_names()

# Load the SEC data
sec <-
  read.csv("../data/data-raw/sec_im9higam.csv", header = TRUE) %>%
  janitor::clean_names()

```

```{r data-tidying}
standards <- standards %>%
    rename(
        c(
            "Ferritin" = curve_ferritin,
            "IgG" = curve_ig_g,
            "Myoglobin" = curve_myoglobin,
            "Ovalbumin" = curve_ovalbumin,
            "Thyroglobulin" = curve_thyroglobulin,
            "Transferrin" = curve_transferrin,
            "Vitamin B12" = curve_vitamin_b12
        )
    ) %>%
    pivot_longer(cols = !c(x, vol),
                 names_to = "protein",
                 values_to = "absorbance_units")

sec <- sec %>%
    pivot_longer(cols = starts_with("a"),
                 names_to = "protein",
                 values_to = "absorbance")

# Separate the PCR data set by DNA library for analysis
gdna <- pcr %>% # gDNA PCR data
    filter(library == "gDNA")

cdna <- pcr %>% # cDNA
    filter(library == "cDNA")
```

___
# Analysis 
## PCR
```{r pcr-linear-models}
#gDNA
gdna_mod <- lm(data = gdna_calibration, log ~ rf)

#cDNA
cdna_mod <- lm(data = cdna_calibration, log ~ rf)
```

```{r pcr-preperation}
# gDNA
gdna <- gdna %>%
  mutate(
    rf = relative_front(gdna),
    log = ((rf * gdna_mod$coefficients[2]) + gdna_mod$coefficients[1]),
    actual_size = (10 ^ log),
    diff = ((expected_size - actual_size) / expected_size * 100)
  )

# cDNA
cdna <- cdna %>%
  mutate(
    rf = relative_front(cdna),
    log = ((rf * cdna_mod$coefficients[2]) + cdna_mod$coefficients[1]),
    actual_size = (10 ^ log),
    diff = ((expected_size - actual_size) / expected_size * 100)
  )

# join gDNA & cDNA
pcr_full <- full_join(gdna, cdna, by = NULL)
```

```{r pcr-fig, include = TRUE, fig.cap = "(fig:pcr-fig)"}
pcr_plot <- pcr_full %>%
    ggplot(aes(
        x = gene,
        y = diff,
        fill = library,
        text = paste(
            "Library: ",
            library,
            "<br>Gene: ",
            gene,
            "<br>Difference: ",
            round(diff),
            "%",
            "<br>Expected Size (bp): ",
            expected_size,
            "<br>Actual Size (bp):",
            round(actual_size)
        )
    )) + geom_col(position = "dodge") +
    geom_hline(yintercept = 0, size = 0.25) +
    mytheme() +
    scale_fill_brewer(name = "DNA Library",
                      labels = c("cDNA", "gDNA"),
                      palette = "Set3",
    ) +
    labs(y = "Percentage Difference to Expected Size", x = "Gene")


htmltools::div(ggplotly(pcr_plot, tooltip = c("text")), align = "center")
```

Diagnostic PCR was used to amplify the *H. influenzae homologues*, and to test for their transcription relative to the Genomic DNA (gDNA) control. Amplicon sizes were established using DNA ladder molecular weight (MW) standards. Results identified the transcription of HiA, HiB, HiGam, HiA-HiB and HiB-HiGam in *H.
influenzae*. Figure \@(fig:pcr-fig) shows

$$\Large \color{#8D9FCA} y \color{black} = \color{#E789C2} m \color{#66c2a4} x \color{black} + \color{#FFD82F} c$$

---
## Electrophoretic Mobility Shift Assay (EMSA) & Binding Affinity

```{r emsa-prep}
emsa <-
  emsa %>% mutate(rf = round((
    (gel_length - distance) / gel_length
  ), digits = 2))
```

```{r emsa-linear-model}
emsa_calibration <- emsa_calibration %>% mutate(log = log(expected_size), rf = (gel_length - distance)/gel_length)
```

```{r nls}
emsa_fit <- NLS(emsa)
```

```{r extract-pred}
emsa_rf = summary(emsa_fit)$coefficients[1, 1]
emsa_intercept = summary(emsa_fit)$coefficients[2, 1]
```

```{r emsa-diff}
# Calculate the difference between the measured rf and predicted rf value
# This is needed for the residuals
# Calculate residuals to the fit
emsa_res <- emsa %>%
    mutate(residuals = rf - predict(emsa_fit))
```

```{r mw-footprint}
emsa <- emsa %>%
  mutate(log = ((emsa$rf * emsa_rf) + emsa_intercept),
           actual_size = (10^log))

emsa_footprint <- (max((emsa$actual_size) - 330000) / (30626.5 / 2))

```

Electrophoretic mobility shift assay (EMSA) was used to examine whether Im9-HiGam–DNA had an affinity for closed circular DNA or linear DNA. Im9-HiGam was observed to bind preferentially to linear DNA with a KD of approximately `r signif(emsa_intercept, digits = 2)` (\@ref(fig:emsa-fig)). Results estimate Im9-HiGam has a binding dimeric footprint of `r signif(abs(emsa_footprint), digits = 2)` base pairs.

```{r emsa-fig, include = TRUE, fig.cap = "(fig:emsa-fig)"}
emsa_plot <- emsa %>%
    ggplot(aes(x = conc, y = rf)) +
    geom_point(shape = 20,
               size = 3,
               colour = "black") +
    stat_function(fun = pred_intercept_emsa, colour = "#66C2A4") +
    mytheme() +
    scale_y_continuous(expand = c(0.02, 0.02),
                       limits = c(0.3, 0.7),
                       breaks = seq(0.3, 0.7, by = 0.20)) +
    scale_x_continuous(
        expand = c(0.02, 0.02),
        limits = c(0, 4),
        breaks = seq(0, 8, by = 1)
    ) +
    xlab(NULL) +
    ylab(expression(paste(bold(
        "Relative Front (a.U.)"
    ))))

# Plot the residuals and a line showing y = 0
emsa_res_plot <- emsa_res %>%
  ggplot(aes(x = conc, y = residuals)) +
  geom_point(shape = 20,
             size = 3,
             colour = "black") +
  geom_hline(aes(yintercept = 0),
             col = '#66C2A4',
             size = .5) +
  scale_x_continuous(
    expand = c(0.02, 0.02),
    limits = c(0, 4),
    breaks = seq(0, 8, by = 1),
    name = (expression(bold(
      atop("Protein Concentration (μM)", "Im9"["dimer"])
    )))
  ) +
  scale_y_continuous(
    expand = c(0.02, 0.02),
    limits = c(-0.1, 0.1),
    breaks = seq(-0.1, 0.1, by = 0.1),
    name = "Residual (a.U.)"
  ) + mytheme()


ggarrange(emsa_plot, emsa_res_plot, nrow = 2, ncol = 1, widths = c(2, 1/2))
```

___
## Size Exclusion Chromatography (SEC)
```{r mw-ev}
mw_ev <- standards %>%
    group_by(protein) %>%
    filter(absorbance_units == max(absorbance_units)) %>%
    summarise(vol) %>%
    ungroup() %>%
    mutate(
        mw = recode(
            protein,
            "Thyroglobulin" = 669000,
            "Ferritin" = 440000,
            "IgG" = 150000,
            "Transferrin" = 81000,
            "Ovalbumin" = 43000,
            "Myoglobin" = 17000,
            "Vitamin B12" = 1355
        ),
        logmw = log(mw)
    )
```

```{r sec-lm}
linear_reg <- lm(mw_ev$vol ~ mw_ev$logmw)
intercept <- linear_reg$coefficients[1] # intercept value
slope <- linear_reg$coefficients[2] # slope value
```

```{r sec-mw}
# Getting the elution volume that corresponds to the maximum value of a curve
volgam <- sec %>%
  group_by(protein) %>%
  filter(absorbance == max(absorbance))

# Estimated MW from the 280 nm data
a280_mw <-
  round(as.numeric(exp((
    volgam$vol[1] - intercept
  ) / slope)), digits = 2)

# Estimated MW from the 254 nm data
a254_mw <-
  round(as.numeric(exp((
    volgam$vol[2] - intercept
  ) / slope)), digits = 2)

sec_footprint <-
  abs(round(((a254_mw - a280_mw) / 660), digits = 2))
```

Size exclusion chromatography (SEC) was used to explore the molecular weight and oligomeric state of Im9-HiGam. 7 chromatograms of known MW were plotted and a linear regression was performed to determine the relationship between elution volume and molecular weight. The SEC data showed a peak for both wavelengths, with a MW of approximately `r format(a254_mw, scientific = FALSE)` Daltons for A254 and `r format(a280_mw, scientific = FALSE)` Daltons for A280. This gives Im9-HiGam has an estimated dimeric footprint of approximately `r format(sec_footprint, scientific = FALSE)` base pairs.

### {.tabset}
#### SEC analysis {-}
```{r sec-fig, include = TRUE, fig.cap= "(fig:sec-fig)"}
sec_plot <- sec %>%
    ggplot(aes(
        x = vol,
        y = absorbance,
        colour = protein),
    show.legend = TRUE) +
    geom_line() +
    scale_colour_brewer(name = "Protein",
                        labels = c("a280", "a254"),
                        palette = "Set2") +
    scale_x_continuous(limits = c(13, 17)) +
    scale_y_continuous(expand = c(0, 0),
                       limits = c(0, 100)) +
    mytheme() +
    xlab("Elution Volume (mL)") +
    ylab("Absorbance (mAU)")


htmltools::div(ggplotly(sec_plot, align = "center"))
```
#### SEC Standards {-}
```{r standards-plot, include = TRUE}
standards_plot <- standards %>%
    ggplot(aes(
        x = vol,
        y = absorbance_units,
        colour = protein),
    show.legend = TRUE) +
    geom_line() +
    scale_colour_brewer(name = "Protein",
                        palette = "Set2") + # colour blind friendly
    mytheme() +
    scale_x_continuous(
        expand = c(0, 0),
        limits = c(9, 22),
        breaks = seq(9, 22, by = 1),
        name = "Elution Volume (mL)"
    ) + scale_y_continuous(
        expand = c(0, 0),
        limits = c(0, 100),
        breaks = seq(0, 100, by = 10),
        name = "A280 Absorbance Units (A.U.)"
    )


htmltools::div(ggplotly(standards_plot, align = "center"))
```

#### SEC dataset {-}
```{r sec-tab, include = TRUE}
sec[2:4] %>%
  pivot_wider(names_from = "protein",
              values_from = "absorbance") %>% 
  knitr::kable(
    caption = "SEC Absorbencies for a254 & a280",
    digits = 2,
    col.names = c("Volume",
                  "a254",
                  "a280")
  ) %>%
  kable_styling(font_size = 15) %>% 
  collapse_rows(columns = 1)
```

# {-}
___
**This Document:** `r count_words("report.Rmd")`\
**README:** `r count_words("../README.md")`\
**Total:** `r count_words("report.Rmd") + count_words("../README.md")`\

___
Report made with `rmarkdown` `r Cite(pkgs, c("rmarkdown","rmarkdown2018", "rmarkdown2020"))`, `RefManageR``r Cite(pkgs, "RefManageR")`.\

Theme is *Minty* by [Bootswatch](https://bootswatch.com/minty/).

___
# Bibliography {.tabset}

## Packages {-}
```{r pkgs-bib, include = TRUE, results = "asis"}
RefManageR::PrintBibliography(pkgs)
```

## References {-}
```{r refs-bib, include = TRUE, results =  "asis"}
RefManageR::PrintBibliography(refs)
```
